### PR TITLE
Worker: Add tests for download_and_extract_zip_file

### PR DESCRIPTION
**Pull request by nikochiko**
As a Google Code-In task.

Changes proposed:
* Move some part of `download_and_extract_zip_file` into  `extract_zip_file` and `delete_zip_file`
* Add tests for the function `download_and_extract_zip_file` and the two new helpers.

### CODE REVIEW

File: scripts/workers/submission_worker.py [hunk 1]
```diff
@@ -130,6 +130,35 @@ def download_and_extract_file(url, download_location):
                     f.write(chunk)
 
 
+def extract_zip_file(download_location, extract_location):
```
Comment (Ram81) [inline scripts/workers/submission_worker.py]: @nikochiko this is a great work on modularizing the code for submission worker. But it would be good to have this in a separate PR. We can create a GCI task for the same. Can you please remove these changes from this PR and open a new PR

Comment (nikochiko) [reply scripts/workers/submission_worker.py]: Sure! Will do that.

**Comment by nikochiko** [comment]
@Ram81 Sorry for making unmentioned changes. I thought it would be weird to have tests that mock the specific modules used (zipfile / os) and test each time they are executed. So I refactored the part so that the tests would be compatible even if we change the dependencies / small part of code. 

Then for now should I edit the tests to make them compatible with the current code or wait for the new PR with refactoring to be merged?

File: scripts/workers/submission_worker.py (hunk 1 above)
Comment (nikochiko) [reply scripts/workers/submission_worker.py]: @Ram81 The new PR: https://github.com/Cloud-CV/EvalAI/pull/2532 
Please have a look.

**Comment by yashdusing** [comment]
@nikochiko it would be better if you could separate the test cases for the newly added functions into #2532

**Comment by nikochiko** [comment]
@yashdusing Done that! 👍 😄

File: tests/unit/worker/test_submission_worker.py [hunk 2]
```diff
@@ -220,3 +226,81 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
...
+        with open(self.download_location, "wb") as zf:
+            zf.write(self.zip_file.getvalue())
+
+        extract_zip_file(self.download_location, self.extract_location)
+        extracted_path = join(self.extract_location, self.file_name)
+        self.assertTrue(os.path.exists(extracted_path))
+        with open(extracted_path, "rb") as extracted:
+            self.assertEqual(extracted.read(), self.file_content)
+
+    def test_delete_zip_file(self):
```
Comment (Ram81) [inline tests/unit/worker/test_submission_worker.py]: @nikochiko I don't think we will need this test as it is already there in #2532

File: tests/unit/worker/test_submission_worker.py [hunk 3]
```diff
@@ -220,3 +226,81 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
...
+        with open(self.download_location, "wb") as zf:
+            zf.write(self.zip_file.getvalue())
+
+        delete_zip_file(self.download_location)
+
+        self.assertFalse(os.path.exists(self.download_location))
+
+    @mock.patch("scripts.workers.submission_worker.logger.error")
+    @mock.patch("scripts.workers.submission_worker.os.remove")
+    def test_delete_zip_file_error(self, mock_remove, mock_logger):
```
Comment (Ram81) [inline tests/unit/worker/test_submission_worker.py]: @nikochiko I don't think we will need this test as it is already there in #2532

File: tests/unit/worker/test_submission_worker.py [hunk 4]
```diff
@@ -220,3 +226,81 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
...
+        responses.add(
+            responses.GET, self.req_url,
+            body=Exception(e))
+        error_message = "Failed to fetch file from {}, error {}".format(self.req_url, e)
+
+        download_and_extract_zip_file(self.req_url, self.download_location, self.extract_location)
+
+        mock_logger.assert_called_with(error_message)
+
+    def test_extract_zip_file(self):
```
Comment (Ram81) [inline tests/unit/worker/test_submission_worker.py]: @nikochiko I don't think we will need this test as it is already there in #2532

File: scripts/workers/submission_worker.py (hunk 1 above)
Comment (pushkalkatara) [reply scripts/workers/submission_worker.py]: @nikochiko we can revert these changes in this PR, as #2532 has the same change.

Comment (nikochiko) [reply scripts/workers/submission_worker.py]: @pushkalkatara the tests written are dependent on these functions 😕

Comment (nikochiko) [reply scripts/workers/submission_worker.py]: Okay I think we can afford the Travis build failing for a few hours. Was worried for no reason, will make the changes :+1:

**Comment by nikochiko** [comment]
@pushkalkatara Done! :+1:

**Comment by yashdusing** [comment]
@nikochiko please check the travis build

**Comment by nikochiko** [comment]
@yashdusing The Travis build is failing due to unmerged dependent changes. I moved the new functions to the other PR upon suggestion of the mentors.
It will be fixed once the other PR (https://github.com/Cloud-CV/EvalAI/2532) is merged.

File: tests/unit/worker/test_submission_worker.py [hunk 5]
```diff
@@ -220,3 +224,53 @@ def test_get_or_create_sqs_queue_for_non_existing_queue(self):
...
+        with zipfile.ZipFile(self.zip_file, mode="w", compression=zipfile.ZIP_DEFLATED) as zipper:
+            zipper.writestr(self.file_name, self.file_content)
+
+    def tearDown(self):
+        if os.path.exists(self.extract_location):
+            shutil.rmtree(self.extract_location)
+
+    @responses.activate
+    @mock.patch("scripts.workers.submission_worker.delete_zip_file")
+    @mock.patch("scripts.workers.submission_worker.extract_zip_file")
```
Comment (yashdusing) [inline tests/unit/worker/test_submission_worker.py]: @nikochiko since this is creating the travis issue maybe you should move this as well to #2532

Comment (nikochiko) [reply tests/unit/worker/test_submission_worker.py]: @yashdusing I don't think this change will go with the purpose of the PRs as the main objective of this PR is to write these tests. As the other PR is approved, I think we can wait for its merge. 
Even if we edit the tests to work, it will be very temporary or the tests will be very sensitive to the specific modules used.
What do you think?

**Comment by RishabhJain2018** [comment]
@nikochiko Can you please move all the changes for this PR into #2532 and close this PR.

**Comment by nikochiko** [comment]
@Ram81 I have moved the changes to #2531 upon the suggestion of @RishabhJain2018 
The PR is already approved, so if the code is good can I get the same on the GCI dashboard?
Also, please create the new task for refactoring or let me know if that has been done already. I will claim that as soon as this task is approved! :D
